<launch>
    <!-- Start ROSBridge WebSocket server -->
    <node name="rosbridge_websocket" pkg="rosbridge_server" type="rosbridge_websocket" output="screen">
        <param name="port" value="9090" />
    </node>

    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
        <rosparam command="load" file="$(find megaProject/config/ekf.yaml)" />
        <remap from="odometry/filtered" to="odom" />
    </node>

    <!-- Start the movement_control node -->
    <node name="movement_control" pkg="mega_project_pkg" type="movement_control" output="screen">
    </node>
    <!-- start spi node -->
    <node name="spi_node" pkg="mega_project_pkg" type="spi_node.py">
    <param name="spi_bus" value="0"/>
    <param name="spi_device" value="0"/>
    <param name="spi_speed" value="1000000"/>
    <param name="rate" value="10"/>
</node>
</launch>
